src/dscanner/analysis/if_constraints_indent.d

```diff
@@ -4,194 +4,132 @@
 
 module dscanner.analysis.if_constraints_indent;
 
-import dparse.lexer;
-import dparse.ast;
 import dscanner.analysis.base;
-import dsymbol.scope_ : Scope;
-
-import std.algorithm.iteration : filter;
-import std.range;
+import dmd.tokens : Token, TOK;
 
 /**
 Checks whether all if constraints have the same indention as their declaration.
 */
-final class IfConstraintsIndentCheck : BaseAnalyzer
+extern (C++) class IfConstraintsIndentCheck(AST) : BaseAnalyzerDmd
 {
+	alias visit = BaseAnalyzerDmd.visit;
 	mixin AnalyzerInfo!"if_constraints_indent";
 
-	///
-	this(BaseAnalyzerArguments args)
-	{
-		super(args);
+	private enum string KEY = "dscanner.style.if_constraints_indent";
+	private enum string MSG = "If constraints should have the same indentation as the function";
 
-		// convert tokens to a list of token starting positions per line
+	private Token[] tokens;
 
-		// libdparse columns start at 1
-		foreach (t; tokens)
-		{
-			// pad empty positions if we skip empty token-less lines
-			// t.line (unsigned) may be 0 if the token is uninitialized/broken, so don't subtract from it
-			// equivalent to: firstSymbolAtLine.length < t.line - 1
-			while (firstSymbolAtLine.length + 1 < t.line)
-				firstSymbolAtLine ~= Pos(1, t.index);
-
-			// insert a new line with positions if new line is reached
-			// (previous while pads skipped lines)
-			if (firstSymbolAtLine.length < t.line)
-				firstSymbolAtLine ~= Pos(t.column, t.index, t.type == tok!"if");
-		}
-	}
-
-	override void visit(const FunctionDeclaration decl)
+	extern (D) this(string fileName, bool skipTests = false)
 	{
-		if (decl.constraint !is null)
-			checkConstraintSpace(decl.constraint, decl.name);
+		super(fileName, skipTests);
+		lexFile();
 	}
 
-	override void visit(const InterfaceDeclaration decl)
+	private void lexFile()
```

Comment:
I have seen this method before. Maybe it should be made public and let all visitors use it.